File: LV. 1/hate_same_num.py
# def solution(arr):
#     answer = []
#     j = -1
#     for i in arr:
#         if i != j:
#             j = i
#             answer.append(j)
#     return answer


# print(solution([1, 1, 3, 3, 0, 1, 1]))


def no_continuous(s):
    a = []
    for i in s:
        if a[-1:] == [i]:
            continue
        a.append(i)
    return a


print(no_continuous([1, 1, 3, 3, 0, 1, 1]))

File: JS lv1/hateSameNum.js
//my code
function solution(arr) {
  var answer = [];
  var j = -1;
  for (var i = 0; i < arr.length; i++) {
    if (arr[i] != j) {
      j = arr[i];
      answer.push(j);
    } else continue;
  }
  return answer;
}

console.log(solution([1, 1, 3, 3, 0, 1, 1]));

//short code
function solution(arr) {
  return arr.filter((val, index) => val != arr[index + 1]);
}
console.log(solution([1, 1, 3, 3, 0, 1, 1]));
